hera_cal/abscal.py

```diff
@@ -632,60 +793,216 @@ def TT_Phi_gain_arr(self):
+    a.add_argument("--delay_cal", default=False, action='store_true', help='perform antenna delay calibration')
+    a.add_argument("--avg_phs_cal", default=False, action='store_true', help='perform antenna avg phase calibration')
+    a.add_argument("--delay_slope_cal", default=False, action='store_true', help='perform delay slope calibration')    
+    a.add_argument("--abs_amp_cal", default=False, action='store_true', help='perform absolute amplitude calibration')
+    a.add_argument("--TT_phs_cal", default=False, action='store_true', help='perform Tip-Tilt phase slope calibration')
+    a.add_argument("--gen_amp_cal", default=False, action='store_true', help='perform general antenna amplitude bandpass calibration')
+    a.add_argument("--gen_phs_cal", default=False, action='store_true', help='perform general antenna phase bandpass calibration')
+    return a
+
+
+def abscal_run(data_files, model_files, pol_select=None, verbose=True, overwrite=False,
+               write_calfits=True, calfits_fname=None, return_gains=False, outdir=None,
+               match_red_bls=False, tol=1.0, reweight=False, interp_model=True, alt_gains=False,
+               delay_slope_cal=True, abs_amp_cal=True, TT_phs_cal=True,
+               delay_cal=False, avg_phs_cal=False, gen_amp_cal=False, gen_phs_cal=False):
+    """
+    run AbsCal on a set of time-contiguous data miriad files, using 
+    time-contiguos model miriad files that cover the data_files across LST
 
+    Parameters:
+    -----------
+    data_files : type=list or string, path(s) to data miriad file(s)
+                a list of paths to miriad file(s) containing complex
+                visibility data, or a path itself
 
-        Parameters:
-        -----------
-        data : 
+    model_files : type=list or sring, path(s) to model miriad files(s)
+                a list of paths to miriad file(s) containing complex
+                visibility data, or a path itself
 
+    verbose : type=boolean, if True print output to stdout
 
-        flags : 
-    
-        """
-        # configure parameters
+    pol_select : type=list, list of polarization strings to use. Ex ['xx']
+
+    overwrite : type=boolean, if True, overwite output files
+
+    write_calfits : type=boolean, if True, write out gains as calfits file
 
+    calfits_fname : type=str, filename of output calfits filename
 
+    outdir : type=str, path to output directory
 
-        Ntimes
-        Nfreqs
+    return_gains : type=boolean, if True, return AbsCal gain dictionary
 
-        # interpolate flagged data
+    match_red_bls : type=boolean, match unique data baselines to model baselines based on redundancy
 
+    reweight : type=boolean, reweight unique baseline data based on redundancy
 
-        # sphere training data x-values
+    interp_model : type=boolean, interpolate model freq and time onto data freq and time
 
+    alt_gains : type=boolean, if True, use full antenna list in data file to make gains,
```

Comment:
I find this description confusing. What exactly is the use case here?

Reply:
for a `omnical` model visibility data. The antennas present only in the omnical model data are only those that form unique baselines, but the file still contains information on all of the antennas actually present in the array. When we want to actually create a gain solution to apply to the data, we want to form a gain solution using all of these antennas, not just the ones present in the data.

Reply:
Ok..."alt gains" is a strange name for this...perhaps we can brainstorm a better one.

Reply:
changed to `full_antenna_gains`, which is more inline with what its doing